source-http-ingest/src/server.rs

```diff
@@ -127,6 +129,18 @@ pub async fn run_server(
     Ok(())
 }
 
+/// Handles GET requests, in order to satisfy Okta's requirements for webhooks.
+/// See issue: https://github.com/estuary/connectors/issues/2433
+/// The TLDR is that we need to take the value of the `x-okta-verification`
+/// header and return it in a JSON response. I have no idea why they do this.
+async fn handle_get(
+    OktaVerificationHeader(okta_header): OktaVerificationHeader,
```

Comment:
This is kind of a nit-pick since the end result is the same and there's nothing else involved in serving a `GET` request... but:

I wasn't expecting  the `OktaVerificationHeader` extractor to also be responsible for returning the 404 if it's not provided, rather I would have expected an `Option<OktaVerificationHeader>` and for the route handler to decide what to do if it's `None`. Looks like this is [supported](https://docs.rs/axum/latest/axum/extract/index.html#optional-extractors) via `OptionalFromRequestParts`. Thoughts on this?

Reply:
The way I see it, `handle_get` _is_ deciding what to do if the header is missing, by declaring the function parameter as just `OktaVerificationHeader`. I guess it doesn't decide what the response body should be, but 🤷. We can change the parameter type later to an `Option` or `Result` if we want to handle GET requests where that header is missing, but for now we don't have a need.